managers/views.py

```diff
@@ -1,8 +1,36 @@
+from django.db.models import Count
+from rest_framework import renderers
 from rest_framework import viewsets
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from users.models import CustomUser
 from managers.models import Project
 from managers.serializers import ProjectSerializer
 
 
 class ProjectViewSet(viewsets.ModelViewSet):
     queryset = Project.objects.all()
     serializer_class = ProjectSerializer
+
+
+class ProjectsList(APIView):
+    renderer_classes = [renderers.TemplateHTMLRenderer]
+    template_name = 'managers/projects_list.html'
+
+    def get_queryset(self):
+        return Project.objects.all().order_by('id')
+
+    def get(self, request):
+        projects_queryset = self.get_queryset()
+        if request.user.user_type == CustomUser.UserType.MANAGER.name:
+            projects_queryset = Project.objects.filter(managers__id=request.user.pk)
+        if request.GET.get('sort'):
```

Comment:
You can define sorting using pure rest framework functionality https://www.django-rest-framework.org/api-guide/filtering/#orderingfilter . 

As for `members` and `members_count` it will be tricky, because defined items in `ordering_fields` must exists on queryset results. But on the other hand, we don't want to run `annotate(members_count=Count('members')` if it is not needed, as it results in decent extra SQL work. So you need to check `if 'members' in request.GET.get('sort')` in `get_queryset` method, and then add counting only then.

Reply:
Done